src/masternodes/poolpairs.h

```diff
@@ -286,17 +286,26 @@ class CPoolPairView : public virtual CStorageView
                 // distribute trading fees
                 if (pool.swapEvent) {
                     CAmount feeA = pool.blockCommissionA * liqWeight / PRECISION;       // liquidity / pool.totalLiquidity;
+                    if (newRewardCalc) {
+                        feeA = pool.blockCommissionA * liquidity / pool.totalLiquidity;
+                    }
                     distributedFeeA += feeA;
                     onTransfer(provider, {pool.idTokenA, feeA}); //can throw
 
                     CAmount feeB = pool.blockCommissionB * liqWeight / PRECISION;       // liquidity / pool.totalLiquidity;
+                    if (newRewardCalc) {
```

Comment:
I feel it would actually be better the other way round. `if (!newRewardCalc)` because the main flow should be the default flow and the condition is for the deprecated calc.

It's just minor code styling, if you can fix it it woudl be best, if not, not a blocker.

Reply:
Updated.